frontend/src/metabase/query_builder/reducers.js

```diff
@@ -196,12 +197,20 @@ export const uiControls = handleActions(
       next: (state, { payload }) => ({ ...state, isEditing: false }),
     },
 
-    [RUN_QUERY]: state => ({ ...state, isRunning: true }),
+    [RUN_QUERY]: state => ({
+      ...state,
+      isRunning: true,
+      isQueryComplete: false,
+    }),
     [CANCEL_QUERY]: {
       next: (state, { payload }) => ({ ...state, isRunning: false }),
     },
     [QUERY_COMPLETED]: {
-      next: (state, { payload }) => ({ ...state, isRunning: false }),
+      next: (state, { payload }) => ({
+        ...state,
+        isRunning: false,
+        isQueryComplete: true,
+      }),
```

Comment:
Why do we need another flag to track the same state? 🤔

Reply:
If it's for readability, should we maybe just add a selector like:

```js
const getIsQueryComplete = state => !state.qb.isRunning
```

Reply:
Oh, got it! Should we maybe replace them with an enum status field? Something like:

```js
type QueryStatus = "idle" | "running" | "complete"

// selectors

const getIsRunning = state => state.qb.queryStatus === "running"
const getIsQueryComplete = state => state.qb.queryStatus === "complete"
```

When we have a few flags in state/reducer reflecting different states of the same process, it's easier to accidentally break them. You can update one flag, but forget to update another one, so the state can become inconsistent. Using a single enum can make it more reliable. WDYT?

Reply:
I like it! Should also simplify the useEffect hook too. I'll see what I can do